src/webgl/p5.Shader.js

```diff
@@ -382,6 +382,18 @@ p5.Shader = class {
    * canvas toggles between a circular gradient of orange and blue vertically. and a circular gradient of red and green moving horizontally when mouse is clicked/pressed.
    */
   setUniform(uniformName, data) {
+    let other = this._renderer.filterShader;
```

Comment:
Maybe I'm not understanding, but isn't this unnecessary? I think you already set the uniform in the filter function. If it is necessary, can we check if the filter layer exists first before executing the code here?

Reply:
Another thought, what about a case where you have two of the same shader, but want to pass different uniforms to them, or use them in different contexts throughout your sketch, (for ex an invert shader running on both a piece of geometry in the main sketch and an invert filter for the filter function) Would this code support that behavior?

Reply:
Great point, I didn't think of that use case. 

On transferring uniforms from the main shader to the secondary shader: A better way of detecting when to copy uniforms might be checking for a child / copy reference, something like `if this.has duplicateShader, then this.duplicateShader.setUniform(...) instead`. And that reference would be set when the duplicateShader is created. I think that should work - not sure why I thought of this other way first.

The whole thing feels a little icky though - first having to create another webgl graphics context, and now having to duplicate shaders since the user's `createShader()` call is bound to the main renderer, and there doesn't seem like a decent way to rebind an existing shader to a different webgl context. And I think there's a little overhead to compiling another shader, although it's been a negligible effect in my testing. I think this duplication complexity would be avoided with the framebuffer approach.

Reply:
The overhead to create and compile a single other shader should be minimal, I don't think we need to worry too much since it's ideally only happening once in the lifetime of a sketch